### DISCUSSION TITLE
Azoteq TPS43 trackpad holder

### DISCUSSION THREAD

**Discussion by george-norton** [snsten/Klein#19] [Show and tell]
I have made a holder for the Azoteq TPS43 trackpad. This trackpad is not yet supported by QMK, but a [driver is under development](https://github.com/daskygit/qmk_firmware/blob/pr/multiple_sensors/drivers/sensors/azoteq_iqs5xx.c). It is quite a bit larger than the 34mm cirque, but can be mounted in the same space. The trackpad also supports multitouch so once the driver is complete it should support some nicer gestures than the cirque. I haven't found a source for the official mylar surface for the trackpad, so I have used vinyl wrap instead. I can share the holder freecad files if there is any interest.

![tps43-1](https://github.com/snsten/Klein/assets/30636555/10f1c592-7707-49c0-a836-318464e6b43a)
![MVIMG_20231002_171751](https://github.com/snsten/Klein/assets/30636555/8cd9bc90-d139-4703-bd00-e8a1d5ca6654)
![MVIMG_20231002_171759](https://github.com/snsten/Klein/assets/30636555/6449ea82-ab81-440c-91b7-851c5d803ed0)

**Comment by snsten** [comment]
Somehow missed this, it looks awesome. 
Please feel free to raise a pr if you can for the holder, would love to try it.

**Reply by george-norton** [reply]
Hello, sure I am currently away, but I will raise a PR next week. I used my klein PCBs to experiment with some keyboard things I haven't really played around with. I also have a ZMK Xiao BLE build with a cirque on one half and encoder on the other.

**Reply by george-norton** [reply]
For the Azoteq, I have been doing some QMK USB digitizer development. I built a big trackpad (peacock) with real multi touch support. But my QMK development branch also includes multi touch support for the Azoteq. So if you build a Klein with an Azoteq, and build firmware from my multitouch_experiment branch, you can get proper 5 finger gesture support (scroll, tap, pinch, rotate, etc..) - although realistically there tps43 is only big  enough for 2 fingers.
